src/cmd/build.rs

```diff
@@ -430,6 +430,23 @@ fn do_optimization(
     Ok(())
 }
 
+/// Asserts that the contract's dependencies are compatible to the ones used in ink!.
+///
+/// This function utilizes `cargo tree`, which takes semver into consideration.
+/// Hence this function only returns an `Err` if it is a proper mismatch in the
+/// major version, and not if the versions mismatch on the minor/patch version.
```

Comment:
Hm at the moment `scale-info` is < `1.0` so breaking changes might come in minor version bumps.

Reply:
Well then the versioning for `scale-info` is broken. Semver is pretty clear about this:

>MAJOR version when you make incompatible API changes,
>MINOR version when you add functionality in a backwards compatible manner, and

Reply:
Come to think about it, I know another tool which then doesn't use semver properly 🙈

Reply:
I mean I could hardcode a version mismatch for `scale-info` if the minor version is not the same? That's pretty hacky, but I guess the versioning scheme for `scale-info` won't change anytime soon.

Reply:
> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable

https://semver.org/#spec-item-4

Reply:
We'll go for `1.0` soon, hopefully in time for `ink! 3.0.0` proper

Reply:
Just noticed that `cargo-tree` behaves correcly with `0.y.z`:
```
$ cargo tree --duplicates -i scale-info
error: There are multiple `scale-info` packages in your project, and the specification `scale-info` is ambiguous.
  scale-info:0.3.0
  scale-info:0.6.0
```
So it already detects version mismatches for `0.`.